packages/sdk-browser/src/__clients__/browser-clients.ts

```
@@ -0,0 +1,828 @@
+   * injected into an iframeStamper to create a session that enables both read and write requests.
+   *
+   * @param targetEmbeddedKey
+   * @param expirationSeconds
+   * @param userId
+   * @returns {Promise<SdkApiTypes.TCreateReadWriteSessionResponse>}
+   */
+  loginWithReadWriteSession = async (
+    targetEmbeddedKey: string,
+    expirationSeconds: string = DEFAULT_SESSION_EXPIRATION_IN_SECONDS,
+    userId?: string,
+  ): Promise<SdkApiTypes.TCreateReadWriteSessionResponse> => {
+    const readWriteSessionResult = await this.createReadWriteSession({
+      targetPublicKey: targetEmbeddedKey,
+      expirationSeconds,
+      userId: userId!,
+    });
+
+    // Ensure session and sessionExpiry are included in the object
+    const readWriteSessionResultWithSession = {
+      ...readWriteSessionResult,
+      credentialBundle: readWriteSessionResult.credentialBundle,
+      sessionExpiry: Date.now() + Number(expirationSeconds) * 1000,
+    };
+
+    // store auth bundle in local storage
+    await saveSession(readWriteSessionResultWithSession, this.authClient);
+
+    return readWriteSessionResultWithSession;
+  };
+
+  /**
+   * Logs in with an existing auth bundle. this bundle enables both read and write requests.
+   *
+   * @param credentialBundle
+   * @param expirationSeconds
+   * @returns {Promise<boolean>}
+   */
+  loginWithAuthBundle = async (
+    credentialBundle: string,
+    expirationSeconds: string = DEFAULT_SESSION_EXPIRATION_IN_SECONDS,
+  ): Promise<any> => {
+    try {
+      const whoAmIResult = await this.getWhoami();
+
+      const readWriteSessionResultWithSession = {
+        ...whoAmIResult,
+        credentialBundle: credentialBundle,
+        sessionExpiry: Date.now() + Number(expirationSeconds) * 1000,
+      };
+
+      await saveSession(readWriteSessionResultWithSession, this.authClient);
+      return true;
+    } catch {
+      return false;
```

maybe should throw an error here

---

Added logic to throw an error here if `loginWithAuthBundle` failes